lib/schema.js

```diff
@@ -413,7 +413,7 @@ Schema.prototype._clone = function _clone(Constructor) {
 
   s.tree = clone(this.tree);
   s.paths = Object.fromEntries(
-    Object.entries(this.paths).map(([key, value]) => ([key, value.clone()]))
+    Object.entries(this.paths).map(([key, value]) => ([key, !value.$isMongooseDocumentArray ? value.clone() : value]))
```

Comment:
While this does work with all tests passing, I think this undercuts the core idea of `clone()`: skipping cloning doc arrays seems incorrect.

Maybe try adding `_skipApplyDiscriminators` in the DocumentArray schema file like here? https://github.com/Automattic/mongoose/pull/14011/files#diff-6f3b287d6939d72b68c31e5ccd577a699f106a13e3068bc977ff92dc13b9d934